server/src/main/java/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1286,12 +1288,31 @@ public Host maintain(final PrepareForMaintenanceCmd cmd) {
             throw new InvalidParameterValueException("Unable to find host with ID: " + hostId + ". Please specify a valid host ID.");
         }
 
-        if (_hostDao.countBy(host.getClusterId(), ResourceState.PrepareForMaintenance, ResourceState.ErrorInMaintenance) > 0) {
-            throw new InvalidParameterValueException("There are other servers in PrepareForMaintenance OR ErrorInMaintenance STATUS in cluster " + host.getClusterId());
+        final ResourceState hostState = host.getResourceState();
```

Comment:
Same as above use a array/list::contains() and refactor/reuse as an internal check method.